src/main/java/userinterface/UserInterface.java

```diff
@@ -1,7 +1,476 @@
+			}
+		} while (vendor == null);
+
+		return vendor;
+	}
+	/**
+	 * Helper method that get the RAT type from user.
+	 * Returns the RAT type user choose.
+	 * 
+	 * @param input     Scanner instance to get user input
+	 * @return RatType   the RAT type
+	 */
+	private static RatType chooseRATtype(Scanner input) {
+		RatType rat = null;
+		System.out.println("\nPlease choose a RAT type:\n"
+				+ "1.	LTE\n"
+				+ "2.	WCDM\n");
+		do {
+			String subOption = input.next();		
+			switch (subOption) {
+				case "1":
+					rat = "LTE";//Todo: change to RatType enum
+					break;
+				case "2":
+					rat = "WCDMA";//Todo: change to RatType enum
+					break;
+				default:
+					System.out.println("Invalid input, please try again.");
+					continue;
+			}		
+		} while (rat == null);
+
+		return rat;	
+	}
+	
+	
+	/**
+	 * Helper method that get the RF Ports from user.
+	 * Returns a list of RF Ports.
+	 * 
+	 * @param input        Scanner instance to get user input
+	 * @return RfPorts[]   an array of RF ports
+	 */	
+	private static RfPorts[] chooseRfPorts(String ratType, Scanner input) {
+		int noOfRfPorts;
+		if (ratType.equals("LTE")) {
+			noOfRfPorts = RF_PORT_NUMBER.LTE_RF_PORTS_NUMBER.getValue();
+		} else {
+			noOfRfPorts = RF_PORT_NUMBER.WCDMA_RF_PORTS_NUMBER.getValue();
+		}		
+		RfPorts[] ports = new RfPorts[noOfRfPorts];
+		int count = 0;
+		System.out.println(
+				"Please choose " + noOfRfPorts + " RF ports\n"
+				+ "1.	RF_0 = A\n"
+				+ "2.	RF_1 = B\n"
+				+ "3.	RF_2 = C\n"
+				+ "4.	RF_3 = D\n"
+				+ "5.	RF_4 = E\n"
+				+ "6.	RF_5 = F\n"
+				+ "7.	RF_6 = G\n"
+				+ "8.	RF_7 = H\n");
+		do {
+			String subOption = input.next();
+			switch (subOption) {
+				case "1":
+					ports[count++] = RfPorts.RF_O;
+					break;
+				case "2":
+					ports[count++] = RfPorts.RF_1;
+					break;
+				case "3":
+					ports[count++] = RfPorts.RF_2;
+					break;
+				case "4":
+					ports[count++] = RfPorts.RF_3;
+					break;
+				case "5":
+					ports[count++] = RfPorts.RF_4;
+					break;
+				case "6":
+					ports[count++] = RfPorts.RF_5;
+					break;
+				case "7":
+					ports[count++] = RfPorts.RF_6;
+					break;
+				case "8":
+					ports[count++] = RfPorts.RF_7;
+					break;
+				default:
+					System.out.println("Invalid input, please try again.");
+					continue;		
+			}
+			if (count != noOfRfPorts) {
+				System.out.println("Enter next number:");            						
+			}
+		} while(count < noOfRfPorts);
+		return ports;
+	}
+	/**
+	 * helper method that get the frequency band from user
+	 * returns the frequency band user choose.
+	 * 
+	 * @param input                 Scanner instance to get user input
+	 * @return CarrierFrequencies   Carrier frequency band
+	 */
+	private static CarrierFrequencies chooseFreqBand(String ratType, Scanner input) {
+		CarrierFrequencies freqBand = null;
+		System.out.println("Please choose Carrier Frequency Band:");
+		if (ratType == "LTE") {
+			System.out.println(
+					  "1.	LTE_BAND_1  = 1920 MHz\n"

```

Comment:
I'm not super familiar with the carrier frequency yet, but would you be able replace these with the enum classes in the future (as a TODO item)?